daphne/src/roles.rs

```diff
@@ -239,6 +245,11 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
             return Err(DapAbort::BatchOverlap);
         }
 
+        // Check whether the DAP version in the request matches the one in task config.
```

Comment:
What happens here if the `CollectReq` has a different wire format between two versions?

Reply:
Should we leave a note that says: "Decode `CollectReq` depending on the protocol version"?